hoomd/write/gsd.py

```diff
@@ -203,6 +225,12 @@ def _attach_hook(self):
 
         self._cpp_obj.log_writer = self.logger
 
+        # Maintain a list of open gsd writers
+        weak_writer = weakref.ref(self)
+        _open_gsd_writers.append(weak_writer)
+        self._finalizer = weakref.finalize(self, _finalize_gsd, weak_writer,
+                                           self._cpp_obj),
```

Comment:
It is possible to do this in `__del__` instead of using `weakref.finalize`?

Reply:
Yes. However the finalizer is more flexible for future development. For example if we were to add a `close` method to `GSD`, the finalizer is a cleaner solution. See this example for an explanation: https://docs.python.org/3/library/weakref.html#comparing-finalizers-with-del-methods